search/src/main/java/com/connexta/search/index/IndexManager.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.index;
+
+import com.connexta.search.index.exceptions.IndexException;
+import java.io.InputStream;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+public interface IndexManager {
+
+  void index(
+      @NotBlank String productId, @NotBlank String mediaType, @NotNull InputStream inputStream)
```

Comment:
The stream is going to get converted into a string in the implementation. Might as well pass the string in directly. The string is human readable (in the debugger) and the stream is not.

Reply:
Right now, we just convert it to a string. But in the future, it's going to be json or xml or something else. Wouldn't it make sense to keep it as a InputStream so we're better set up to parse it later?

Reply:
I can buy that. :-)